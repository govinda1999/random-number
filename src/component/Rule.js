import React from 'react'

function Rule(){
	return(
		<>
			<p className="h4 mt-3 pt-3">Rules:</p>
	        <ul className="text-capitalize">
	        <li>Enter number between 1-100.</li>
	        <li>if Difference is zero => Success.</li>
	        <li>if Difference is less than 4 => Hot.</li>
	        <li>if Difference is less than 15 => Warm.</li>
	        <li>if Difference is greater than 15 => Cold.</li>
	        </ul>
		</>
	)
}
export default Rule;