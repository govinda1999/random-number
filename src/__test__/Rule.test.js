import React from 'react';
import { unmountComponentAtNode,render} from "react-dom";
import Rule from '../component/Rule';
import { act } from "react-dom/test-utils";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('Rule component render properly',()=>{
	act(() => {
		render(<Rule />, container);  
	});
	expect(container.textContent)
	.toBe("Rules:Enter number between 1-100.if Difference is zero => Success.if Difference is less than 4 => Hot.if Difference is less than 15 => Warm.if Difference is greater than 15 => Cold.");

	act(() => {
		render(<Rule />, container);  
	});
	expect(container.textContent)
	.not.toBe("Rules:if Difference is zero => Success.if Difference is less than 4 => Hot.if Difference is less than 15 => Warm.if Difference is greater than 15 => Cold.")
})

