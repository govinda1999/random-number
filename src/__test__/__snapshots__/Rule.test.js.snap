// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rule component render properly 1`] = `
"<p class=\\"h4 mt-3 pt-3\\">Rules:</p>
<ul class=\\"text-capitalize\\">
  <li>Enter number between 1-100.</li>
  <li>if Difference is zero =&gt; Success.</li>
  <li>if Difference is less than 4 =&gt; Hot.</li>
  <li>if Difference is less than 15 =&gt; Warm.</li>
  <li>if Difference is greater than 15 =&gt; Cold.</li>
</ul>"
`;
